Source/options.cpp

```diff
@@ -1463,22 +1464,35 @@ void KeymapperOptions::KeyPressed(uint32_t key) const
 	action.actionPressed();
 }
 
-void KeymapperOptions::KeyReleased(uint32_t key) const
+void KeymapperOptions::KeyReleased(SDL_Keycode key) const
 {
+	if (key >= SDLK_a && key <= SDLK_z) {
+		key = static_cast<SDL_Keycode>(static_cast<Sint32>(key) - ('a' - 'A'));
+	}
 	auto it = keyIDToAction.find(key);
 	if (it == keyIDToAction.end())
 		return; // Ignore unmapped keys.
 
 	const Action &action = it->second.get();
 
-	// Check that the action can be triggered and that the chat textbox is not
-	// open.
-	if (!action.actionReleased || (action.enable && !action.enable()) || talkflag)
+	// Check that the action can be triggered and that the chat or gold textbox is not
+	// open. If either of those textboxes are open, only return if the key can be used for entry into the box
+	if (!action.actionReleased || (action.enable && !action.enable()) || ((talkflag && IsTextEntryKey(key)) || (dropGoldFlag && IsNumberEntryKey(key))))
 		return;
 
 	action.actionReleased();
 }
 
+bool KeymapperOptions::IsTextEntryKey(SDL_Keycode vkey) const
+{
+	return ((static_cast<Sint32>(vkey) >= 0 && static_cast<Sint32>(vkey) < 128) || vkey == SDLC_KEYSTATE_LEFTSHIFT || vkey == SDLC_KEYSTATE_RIGHTSHIFT || vkey == SDLK_BACKSPACE);

```

Comment:
`SDLK_BACKSPACE` is 8, so it doesn't need a separate condition. For SDL2, `SDLC_KEYSTATE_LEFTSHIFT` is actually a reference to the scancode, not the keycode. What you want is `SDLK_LSHIFT` which is available in both SDL2 and SDL1.

But you can probably ignore all that because, to remain consistent with `control_presskeys()`, I recommend the following.

```suggestion
	return IsAnyOf(vkey, SDLK_ESCAPE, SDLK_RETURN, SDLK_KP_ENTER, SDLK_BACKSPACE, SDLK_DOWN, SDLK_UP) || (vkey >= SDLK_SPACE && vkey <= SDLK_z);
```

If you want to include left shift and right shift, you should modify `control_presskeys()` as well.